src/flypad-backend/src/Aircraft/AircraftProcedures.h

```diff
@@ -16,285 +19,284 @@ struct ProcedureStep {
   // expected state is already set
   double delayAfter;
   // check if desired state is already set so the action can be skipped
-  std::string expectedStateCheckCode;
+  const char* expectedStateCheckCode;
   // calculator code to achieve the desired state
   // if it is a conditional this calculator code needs to eval to true or false
-  std::string actionCode;
+  const char* actionCode;
 };
 
-// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-// Please remember to also update the EFB Presets page for the step description
-// if you make any changes to this list.
-// src/instruments/src/EFB/Presets/Widgets/AircraftPresets.tsx
-// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-
-// @formatter:off
-static vector<ProcedureStep*>* POWERED_CONFIG_ON = new vector<ProcedureStep*>{
+class AircraftProcedures {
+  // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  // Please remember to also update the EFB Presets page for the step description
+  // if you make any changes to this list.
+  // src/instruments/src/EFB/Presets/Widgets/AircraftPresets.tsx
+  // !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+  static constexpr std::array POWERED_CONFIG_ON {
   // SOP: PRELIMINARY COCKPIT PREPARATION
-  new ProcedureStep {"BAT1 On",                  1010, false, 1000, "(L:A32NX_OVHD_ELEC_BAT_1_PB_IS_AUTO)",                 "1 (>L:A32NX_OVHD_ELEC_BAT_1_PB_IS_AUTO)"},
-  new ProcedureStep {"BAT2 On",                  1020, false, 3000, "(L:A32NX_OVHD_ELEC_BAT_2_PB_IS_AUTO)",                 "1 (>L:A32NX_OVHD_ELEC_BAT_2_PB_IS_AUTO)"},
+  ProcedureStep{"BAT1 On",                  1010, false, 1000, "(L:A32NX_OVHD_ELEC_BAT_1_PB_IS_AUTO)",                 "1 (>L:A32NX_OVHD_ELEC_BAT_1_PB_IS_AUTO)"},
+  ProcedureStep{"BAT2 On",                  1020, false, 3000, "(L:A32NX_OVHD_ELEC_BAT_2_PB_IS_AUTO)",                 "1 (>L:A32NX_OVHD_ELEC_BAT_2_PB_IS_AUTO)"},
 
-  new ProcedureStep {"EXT PWR On",               1030, false, 3000, "(L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON) "
+  ProcedureStep{"EXT PWR On",               1030, false, 3000, "(L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON) "
                                                                     "(L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON) && "
                                                                     "(L:A32NX_ENGINE_STATE:1) 1 == || "
                                                                     "(L:A32NX_ENGINE_STATE:2) 1 == || "
                                                                     "(A:EXTERNAL POWER ON:1, BOOL) ||",                     "(A:EXTERNAL POWER ON:1, BOOL) ! if{ 1 (>K:TOGGLE_EXTERNAL_POWER) }"},
```

Comment:
Please fix the formatting of this file - the line identation and spacing was aligned for a reason

Reply:
That's pretty odd, I think my `clang-format` messed things up here.

Reply:
You need to exclude this file and format manually if you are not using an editor which respects `// @formatter:off` (IDEA CLION does).
And the no format parts need to be done manually anyway. 
It is somewhat a compromise of readability and not having too long lines.